sdk/src/nonce_state.rs

```diff
@@ -0,0 +1,545 @@
+use serde_derive::{Deserialize, Serialize};
+use crate::{
+    account::{Account, KeyedAccount},
+    account_utils::State,
+    hash::Hash,
+    instruction::InstructionError,
+    nonce_program::id,
+    nonce_instruction::{NonceError},
+    pubkey::Pubkey,
+    sysvar::recent_blockhashes::RecentBlockhashes,
+    sysvar::rent::Rent,
+};
+use std::collections::HashSet;
+
+#[derive(Debug, Default, Serialize, Deserialize, PartialEq, Clone, Copy)]
+pub struct Meta {
+    rent_exempt_lamports: u64,
```

Comment:
Is this necessary? I think you could conditionally require the rent account for withdraw instructions where `lamports < account.lamports`

Reply:
The `NonceAccount` itself needs to be rent exempt.  My understanding is that this is signaled by maintaining a minimum balance specified by the network at account creation.

@rob-solana any clarifying comments?

Reply:
Yeah I'm on the same page. Just commenting whether you need to store `rent_exempt_lamports` in the account meta. You only use it when processing withdraw instructions. Instead of storing that meta, you could require the rent account for withdraw instructions which withdraw less than the total lamports

Reply:
@t-nelson what you could do is make `create_nonce_account` instruction require to be passed `AccountMeta` of rent sysvar, and while processing said instruction, just verify that account has sufficient balance, if not just return an error.

You can look up how stake and vote account creation has handled rent for more detailed ref.

Reply:
Thanks!  I actually stole the rent exempt handling from stake 🙃.  Will take another look, though.